plugins/uast-kotlin-fir/src/org/jetbrains/uast/kotlin/declarations/FirKotlinUFile.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.uast.kotlin
+
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.fir.declarations.FirFile
+import org.jetbrains.kotlin.idea.fir.low.level.api.api.getFirFile
+import org.jetbrains.kotlin.idea.fir.low.level.api.api.getResolveState
+import org.jetbrains.kotlin.psi.KtClassOrObject
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.uast.*
+
+class FirKotlinUFile(
+    override val psi: KtFile,
+    override val languagePlugin: UastLanguagePlugin = firKotlinUastPlugin
+) : UFile {
+    override val javaPsi: PsiElement? = null
+
+    override val sourcePsi: KtFile = psi
+
+    // TODO: it's not yet clear whether this is needed
```

Comment:
I guess we won't need this, though, unless facade lookup/creation needs it? @darthorimar any hints/pointers?

Reply:
Let's try to avoid using fir https://github.com/JetBrains/kotlin/pull/4312#issuecomment-824841239 KtFileSymbol may help here

Reply:
I removed the use of FIR here. But, it's still unclear how facade can be retrieved and/or created via KtAnalysisSession.